### PR TITLE
[INLONG-9982][Agent] Adjusting the abstraction of source code to facilitate rapid addition of sources

### LINKED ISSUE #9982: [Improve][Agent] Adjusting the abstraction of source code to facilitate rapid addition of sources
### Description

Further encapsulate the agent source, mask memory control, mainstream process control, and expose the read data interfaces related to specific data sources

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
[INLONG-9982][Agent] Adjusting the abstraction of source code to facilitate rapid addition of sources
- Fixes #9982

### Motivation
To facilitate rapid addition of sources

### Modifications

Further encapsulate the agent source, mask memory control, mainstream process control, and expose the read data interfaces related to specific data sources

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  TestLogfileSource

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

- [] #9983

### CODE REVIEW

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/file/AbstractSource.java [hunk 1]
```diff
@@ -41,11 +96,169 @@ public abstract class AbstractSource implements Source {
...
+    /**
+     * After preparation work, we started to truly read data from the data source
+     */
+    private void putIntoQueue(SourceData sourceData) {
+        if (sourceData == null) {
+            return;
+        }
+        try {
+            boolean offerSuc = false;
+            while (isRunnable() && offerSuc != true) {
```
Comment (aloyszhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/file/AbstractSource.java:241]:
```suggestion
            while (isRunnable() && !offerSuc) {
```

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/LogFileSource.java [hunk 2]
```diff
@@ -148,46 +67,77 @@ public LogFileSource() {
...
+    private List<SourceData> readFromPos(long pos) throws IOException {
+        List<byte[]> lines = new ArrayList<>();
+        List<SourceData> dataList = new ArrayList<>();
+        RandomAccessFile input = new RandomAccessFile(file, "r");
+        bytePosition = readLines(input, pos, lines, BATCH_READ_LINE_COUNT, BATCH_READ_LINE_TOTAL_LEN, false);
+        for (int i = 0; i < lines.size(); i++) {
+            linePosition++;
+            dataList.add(new SourceData(lines.get(i), linePosition));
+        }
+        if (input != null) {
```
Comment (aloyszhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/LogFileSource.java:135]: input is always not null

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/PulsarSource.java [hunk 3]
```diff
@@ -141,183 +76,84 @@ public void init(InstanceProfile profile) {
...
-                }
-                AgentUtils.silenceSleepInSeconds(1);
-            }
+    @Override
+    protected List<SourceData> readFromSource() {
+        List<SourceData> dataList = new ArrayList<>();
+        org.apache.pulsar.client.api.Message<byte[]> message = null;
+        try {
+            message = consumer.receive(0, TimeUnit.MILLISECONDS);
+            offset = message.getSequenceId();
```
Comment (aloyszhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/PulsarSource.java:102]: Offset is not very meaningful for pulsar, if you want to show the position of consuming, 
 I think messageId is better.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/PulsarSource.java [hunk 4]
```diff
@@ -141,183 +76,84 @@ public void init(InstanceProfile profile) {
...
-        if (sourceData == null) {
-            return;
-        }
+    private Consumer<byte[]> getConsumer() {
         try {
-            boolean offerSuc = false;
-            if (queue.remainingCapacity() > 0) {
-                while (isRunnable() && !offerSuc) {
-                    offerSuc = queue.offer(sourceData, 1, TimeUnit.SECONDS);
+            try (Consumer<byte[]> consumer = pulsarClient.newConsumer(Schema.BYTES)
```
Comment (aloyszhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/PulsarSource.java:119]: The consumer will be closed if created in the try-with-resource block.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/PulsarSource.java (hunk 3 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/PulsarSource.java:102]: get, the current offset is actually useless. If we really need to save it in the future, we will use the MessageId as the offset

**Review by aloyszhang — APPROVED** [review]
LGTM

**Review by haifxu — APPROVED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/PulsarSource.java [hunk 5]
```diff
@@ -141,183 +76,101 @@ public void init(InstanceProfile profile) {
...
+            offset = message.getSequenceId();
+        } catch (PulsarClientException e) {
+            LOGGER.error("read from pulsar error", e);
         }
-        return true;
+        if (!ObjectUtils.isEmpty(message)) {
+            dataList.add(new SourceData(message.getValue(), 0L));
+        }
+        try {
+            consumer.acknowledge(message);
```
Comment (luchunliang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/PulsarSource.java:110]: It is need to ack message offset after agent is successful to send data to DataProxy.
It maybe miss data if agent is coredump.

**Review by baomingyu — APPROVED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/PulsarSource.java (hunk 5 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/PulsarSource.java:110]: In the future, this goal will be achieved by saving the messageid

**Review by luchunliang — APPROVED** [review]

**Pull request merged**